### PR TITLE
Selection

### PR DESCRIPTION
**Pull request by zabel-xyz**
Currently, if several files are found, all files are opened, but only the last (not always the same) will be shown.
Now, I list them to the user so he can choose which open

### CODE REVIEW

**Comment by pr0g** [comment]
This looks like an awesome change! Really nice addition!

The one thing I'd change is the search should be exact as opposed to partial (which in hindsight might have been what was causing the other issue you mentioned in the previous PR)

If I have a file called test.h and then several other files called testing.h, tested.h and test_all.cpp - all of those should not be listed when trying to switch. However, if you had test.h, test.cpp, test.inl and test.txt, it would be valid to return those three in the list you display if switching from the first.

So when in test.h, hitting the command switch you'd see test.cpp, test.inl and test.txt but not testing.h, tested.h and test_all.cpp.

I think this was an existing bug but is now more apparent with this new change. Let me check if there's an easy fix...

Thanks again dude! Again a really nice addition and makes the extension much friendlier to use!

**Comment by pr0g** [comment]
Just had a look into this and annoyingly it seems you can't pass a regex or anything more complicated into the vscode.workspace.findFiles call.

This bit of code is super noddy and I imagine there is a much more elegant way to write it using a map/reduce pairing or something but you get the idea.

var exact_files = [];
for (let index = 0, file; index < files.length; index++) {
    file = files[index];
    var fn_and_ext = path.basename(file._path);
    var fn = fn_and_ext.substring(0, fn_and_ext.lastIndexOf('.'));
    if ( fn === fileName ) {
        exact_files.push( file );
    }
}

(can't get the formatting to play ball sorry...)

Basically just filter the returned list to only match the filename exactly, as this is 99 times out of 100 the behaviour you want. Then in the following code, use `exact_files` instead of `files` as you currently are, and you will only see the files listed that match exactly.

Does that make sense?

Thanks!


Tom

**Comment by zabel-xyz** [comment]
I used var files = vscode.workspace.findFiles("**/" + fileName + ".*", "**/" + fileNameAndExtension, 100)
but it will not work with dot character inside filename (e.g.: unit.xxx.pas)

much more elegant version of your code:
   let exact_files = files.filter(file => path.basename(file, path.extname(file)) === fileName);

**Comment by pr0g** [comment]
Ah I see - is that a problem though? Does it need to work with dot character inside filename?

Aah that's sweet! I'm going to steal that code and try it in my local test of this PR. If that works could you add it and I'll merge this in too!

Thanks again man, awesome work! 😄

**Comment by pr0g** [comment]
screen shot 2016-05-09 at 20 21 03

(My addition to your change - if you agree that makes sense - that gives me the behaviour I'd expect)

**Comment by zabel-xyz** [comment]
I only added a condition for not calling showQuickPick() when exact_files is empty.

**Comment by pr0g** [comment]
Hey zabel, sorry for just getting back to you.

I'm not sure I understand sorry, have you made the edits I suggested? Is the pull request updated based on our chat? 

Forgive me I'm not much of a GitHub user as is probably apparent...

Thanks!


Tom

**Comment by zabel-xyz** [comment]
I updated the 2 pull requests accoroding  to our chats...
Thanks
Zabel

**Comment by pr0g** [comment]
That's brilliant! Thanks very much!

I'll merge these changes this weekend and then push an update of the extension, I'll let you know when it's done!

Thanks again! 😃

**Comment by pr0g** [comment]
Just working on this one now but in a bit of merge hell at the moment ;)

Should be in before too long...

**Comment by pr0g** [comment]
Argh I might have messed this up a bit... I have merged the changes, but had to resolve extension.ts and now this doesn't seem to think I have correctly merged it - can you check the changes and let me know if you are you happy for me to close this? Thanks!

**Comment by zabel-xyz** [comment]
It's ok, you can close this request... 
Thanks ! 
Zabel

**Comment by pr0g** [comment]
Cool thanks man,

Not at all! Thank you for contributing some awesome code and improving this little extension! :)

👍
